packages/hardhat-test-helpers/src/chai/matchers/bigNumber.ts

```diff
@@ -0,0 +1,264 @@
+// TODO: consider adding support for the following Chai.Assert properties.
+// Might any of these come "for free" due to supporting other methods? At the
+// very least get them tested.
+//   isAbove
+//   isAtLeast
+//   isBelow
+//   isAtMost
+//   isAt
+//   isNumber
+//   isNotNumber
+//   closeTo
+//   approximately
+
+import { BigNumber as EthersBigNumberType } from "ethers";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { BigNumber as BigNumberJsType } from "bignumber.js";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { default as BNType } from "bn.js";
```

Comment:
My understanding is that since I'm only ever using eg `BigNumberJsType` as a type, and never actually instantiating it, `tsc` doesn't actually translate this `import` statement into a `require()` expression, so no such error can happen.

Indeed, I checked the `bigNumber.js` built from this source file, and there are no `require()` statements corresponding to these `import` statements.

Further, I tried adding in a simple `const b = new BNType(3);` in the file scope, and that *did* induce `tsc` to translate the `BNType` `import` to a `require("bn.js")`, suggesting that my presumption (non-instantiation avoids inducing a `require`) seems to be correct.

Reply:
Oh, ok. I guess it feels weird to me to use a dev dependency in this way, but maybe it's fine. We should use `import type` though to make sure no one uses the value accidentally.

@alcuadrado do you have thoughts about this?

Reply:
Switched to using `import type` in https://github.com/NomicFoundation/hardhat/pull/2428/commits/7d80addfc666eccc02d94ec3e307acf7e86d3420

Reply:
`import type` doesn't get translated to a runtime `require`, so it doesn't fail if not present

Reply:
afaik that's actual the goal of that TS construct, being explicit that you don't want a runtime import.